back-end/src/sciler/handler/handler.go

```diff
@@ -79,24 +79,105 @@ func (handler *Handler) onConnectionMsg(raw Message) {
+				}
+			}
+		case "boolean":
+			{
+				if valueType != reflect.Bool {
+					return fmt.Errorf("input type boolean expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		case "numeric":
+			{
+				if valueType != reflect.Int && valueType != reflect.Float64 {
+					return fmt.Errorf("input type numeric expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		case "array":
+			{
+				if valueType != reflect.Slice {
+					return fmt.Errorf("input type array/slice expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		default:
+			// todo custom types
+			return fmt.Errorf("custom types like: %s, are not yet implemented", inputType)
+		}
+		//} else if inputType, ok := device.Output[k].Type; ok {
+		//	switch inputType {
+		//	case "string":
+		//		{
+		//			if valueType == reflect.String {
+		//				handler.Config.Devices[raw.DeviceID].Status[k] = v
+		//			} else {
+		//				logrus.Warn("input type string expected but ", valueType.String(), " found as type of value ", v)
+		//			}
+		//		}
+		//	case "boolean":
+		//		{
+		//			if valueType == reflect.Bool {
+		//				handler.Config.Devices[raw.DeviceID].Status[k] = v
+		//			} else {
+		//				logrus.Warn("input type boolean expected but ", valueType.String(), " found as type of value ", v)
+		//			}
+		//		}
+		//	case "numeric":
+		//		{
+		//			if valueType == reflect.Int || valueType == reflect.Float64 {
+		//				handler.Config.Devices[raw.DeviceID].Status[k] = v
+		//			} else {
+		//				logrus.Warn("input type numeric expected but ", valueType.String(), " found as type of value ", v)
+		//			}
+		//		}
+		//	case "array":
+		//		{
+		//			if valueType == reflect.Slice {
+		//				handler.Config.Devices[raw.DeviceID].Status[k] = v
+		//			} else {
+		//				logrus.Warn("input type array/slice expected but ", valueType.String(), " found as type of value ", v)
+		//			}
+		//		}
+		//	default:
+		//		// todo custom types
+		//		logrus.Warn("custom types like: ", inputType, ", are not yet implemented")
+		//	}
+	} else {
+		return fmt.Errorf("status message received from component %s, which is not in the config under device %s", k, device.ID)
+	}
+	return nil
+}
+
 //onStatusMsg is the function to process status messages.
 func (handler *Handler) onStatusMsg(raw Message) {
-	_, ok := handler.Config.Devices[raw.DeviceID]
-	if !ok {
-		logrus.Warn("status message received from device " + raw.DeviceID + ", which is not in the config")
-	} else {
-		logrus.Info("status message received from: " + raw.DeviceID + ", status: " + fmt.Sprint(raw.Contents))
+	if device, ok := handler.Config.Devices[raw.DeviceID]; ok {
 		for k, v := range raw.Contents {
-			handler.Config.Devices[raw.DeviceID].Status[k] = v
+			err := handler.checkStatusType(device, v, k)
+			if err != nil {
+				logrus.Warn(err)
+			} else {
+				handler.Config.Devices[raw.DeviceID].Status[k] = v
```

Comment:
can you add the logrus.Info("status message received from: " + raw.DeviceID + ", status: " + fmt.Sprint(raw.Contents)) again